### PR TITLE
Longw/msi azure policy

### PR DESCRIPTION
**Pull request by wanlonghenry**
msi template with azure policy

### CODE REVIEW

File: scripts/onboarding/aks/onboarding-msi-azure-policy/azure-policy.rules.json [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+	  "field": "type",
+	  "equals": "Microsoft.ContainerService/managedClusters"
+	},
+	"then": {
+	  "effect": "deployIfNotExists",
+	  "details": {
+		"type": "Microsoft.ContainerService/managedClusters",
+		"name": "[field('name')]",
+		"roleDefinitionIds": [
+			"/providers/Microsoft.Authorization/roleDefinitions/b24988ac-6180-42a0-ab88-20f7382dd24c"
```
Comment (ganga1980) [inline scripts/onboarding/aks/onboarding-msi-azure-policy/azure-policy.rules.json:12]: you didnt find low privilege built-in roles than contributor?

File: scripts/onboarding/templates/arc-k8s-extension-msi-auth/existingClusterOnboarding.json [hunk 2]
```diff
@@ -260,8 +259,8 @@
       "type": "Microsoft.Resources/deployments",
       "name": "[Concat('arc-k8s-ci-extension', '-',  uniqueString(parameters('clusterResourceId')))]",
       "apiVersion": "2019-05-01",
-      "subscriptionId": "[split(parameters('clusterResourceId'),'/')[2]]",
-      "resourceGroup": "[split(parameters('clusterResourceId'),'/')[4]]",
+      "subscriptionId": "[variables('clusterSubscriptionId')]",
```
Comment (ganga1980) [inline scripts/onboarding/templates/arc-k8s-extension-msi-auth/existingClusterOnboarding.json:262]: I see the same template changes in both PRs and you can revert the changes in this PR and keep other one.

File: scripts/onboarding/templates/arc-k8s-extension-provisionedcluster-msi-auth/existingClusterOnboarding.json [hunk 3]
```diff
@@ -260,8 +259,8 @@
       "type": "Microsoft.Resources/deployments",
       "name": "[Concat('arc-k8s-ci-extension', '-',  uniqueString(parameters('clusterResourceId')))]",
       "apiVersion": "2019-05-01",
-      "subscriptionId": "[split(parameters('clusterResourceId'),'/')[2]]",
-      "resourceGroup": "[split(parameters('clusterResourceId'),'/')[4]]",
+      "subscriptionId": "[variables('clusterSubscriptionId')]",
```
Comment (ganga1980) [inline scripts/onboarding/templates/arc-k8s-extension-provisionedcluster-msi-auth/existingClusterOnboarding.json:262]: similarly, here you can revert these template changes since these changes already covered in other PR.

File: scripts/onboarding/aks/onboarding-msi-azure-policy/README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+You can create the policy definition using a command like :
+
+```az policy definition create --name "CITest" --display-name "CITest" --mode Indexed --metadata version=1.0.0 category=Kubernetes --rules .\azure-policy.rules.json --params .\azure-policy.parameters.json```
+
+**NOTE**
+
+- Please download all files under AddonPolicyTemplate folder before running the policy template.
+- After creating the policy definition through the above command, go to Azure portal -> Policy -> Definitions and select the definition you just created.
+- Click on 'Assign' and then go to the 'Parameters' tab and fill in the details. Then click 'Review + Create'.
```
Comment (ganga1980) [inline scripts/onboarding/aks/onboarding-msi-azure-policy/README.md:9]: For Policy assignment, you can have the instructions similar to this - https://learn.microsoft.com/en-us/azure/azure-monitor/containers/container-insights-enable-aks-policy#assign-a-policy-definition-to-a-specified-scope-1

File: scripts/onboarding/templates/arc-k8s-extension-provisionedcluster-msi-auth/existingClusterOnboarding.json (hunk 3 above)
Comment (wanlonghenry) [reply scripts/onboarding/templates/arc-k8s-extension-provisionedcluster-msi-auth/existingClusterOnboarding.json]: resolved

File: scripts/onboarding/aks/onboarding-msi-azure-policy/azure-policy.rules.json (hunk 1 above)
Comment (wanlonghenry) [reply scripts/onboarding/aks/onboarding-msi-azure-policy/azure-policy.rules.json:12]: after adding those 3 roles https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/data-collection-rule-overview#permissions, the deployment still has error 
image

**Comment by pfrcks** [comment]
can you add the testing results here for the policy?

**Comment by wanlonghenry** [comment]
image

File: scripts/onboarding/aks/onboarding-msi-azure-policy/README.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+You can create the policy definition using a command like in windows:
+
+```az policy definition create --name "Preview-AKS-Monitoring-Addon-MSI" --display-name "Preview-AKS-Monitoring-Addon-MSI" --mode Indexed --metadata version=1.0.0 category=Kubernetes --rules .\azure-policy.rules.json --params .\azure-policy.parameters.json```
```
Comment (jatakiajanvi12) [inline scripts/onboarding/aks/onboarding-msi-azure-policy/README.md:3]: Is it a preview policy? For built-in policies, this is the versioning guide: https://dev.azure.com/msazure/One/_wiki/wikis/One.wiki/87210/Versioning-Preview-Deprecation-Guidelines 
version=1.0.0-preview.

File: scripts/onboarding/aks/onboarding-msi-azure-policy/README.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+For linux or wsl environment, change the backward slash to forward slash.
```
Comment (jatakiajanvi12) [inline scripts/onboarding/aks/onboarding-msi-azure-policy/README.md:5]: The command works without slashes as well in both Windows and wsl, can you check that and change the command itself?

File: scripts/onboarding/aks/onboarding-msi-azure-policy/README.md (hunk 5 above)
Comment (wanlonghenry) [reply scripts/onboarding/aks/onboarding-msi-azure-policy/README.md]: removed

**Review by ganga1980 — APPROVED** [review]
